### PR TITLE
Publish Docker images to Amazon ECR Public

### PR DESCRIPTION
**Pull request by ybiquitous**
> Explain a summary, purpose, or background for this change.

See https://docs.aws.amazon.com/AmazonECR/latest/public/docker-push-ecr-image.html

> Refer related issues or pull requests (e.g. `Close #123` or `None`).

Follow-up of #2465

> Check the following if needed.

- [x] Add a new entry to [CHANGELOG.md](https://github.com/sider/runners/blob/master/CHANGELOG.md) if this change is notable.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -67,13 +67,12 @@ jobs:
           echo "GITHUB_REF: ${GITHUB_REF}"
           tag="${GITHUB_REF#refs/*/}"
           tag="${tag///}"
+          tag=master
```
Comment (ybiquitous) [inline .github/workflows/build.yml:70]: **TODO: REMOVE THIS LINE.**

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -7,6 +7,10 @@ on:
   pull_request:
     branches: ["**"]
 
+env:
+  # TODO: Will change to `public.ecr.aws/sider`.
+  AWS_ECR_REGISTRY: public.ecr.aws/r8f9z5e1
```
Comment (gnakagaw) [inline .github/workflows/build.yml:12]: Can we change to `/sider` now? or later?

Comment (ybiquitous) [reply .github/workflows/build.yml:12]: Maybe later. I have requested the custom alias `sider`, but it does not seem to be approved by AWS yet.

image

Comment (gnakagaw) [reply .github/workflows/build.yml:12]: Got it 👍

**Review by gnakagaw — APPROVED** [review]
🚀

File: .github/workflows/build.yml (hunk 2 above)
Comment (ybiquitous) [reply .github/workflows/build.yml:12]: If we will use the custom alias, do we need to re-push images?

Comment (ybiquitous) [reply .github/workflows/build.yml:12]: Ref: https://docs.aws.amazon.com/AmazonECR/latest/public/public-registry-settings.html

Comment (gnakagaw) [reply .github/workflows/build.yml:12]: > if approved, both the default alias and custom alias can be used to access your public repositories. 

Seems to be OK. But I have a concerns about image metadata. Here is a example of `docker inspect [image]`. It contains `r8f9z5e1` in image URL.. Sorry I'm not confident about this. Let's try it💪 
```
[
    {
        "Id": "sha256:dc5406089013f7817b0a2e58907bc132e7780e70c4aa304bec0b7ede3b8df340",
        "RepoTags": [
            "public.ecr.aws/r8f9z5e1/runner_brakeman:master"
        ],
        "RepoDigests": [
            "public.ecr.aws/r8f9z5e1/runner_brakeman@sha256:bdfd3704fe97c33e65d294aabe9578bf08908f6997e6a7401f30e116171b7839"
        ],
        "Parent": "",
        "Comment": "buildkit.dockerfile.v0",
```

**Comment by ybiquitous** [comment]
Publishing works! 🎉 
https://github.com/sider/runners/actions/runs/959639543

File: .github/workflows/build.yml (hunk 2 above)
Comment (ybiquitous) [reply .github/workflows/build.yml:12]: I’ve created an issue #2469 to fix the TODO.

**Pull request merged**